src/components/Articles.jsx

```diff
@@ -1,21 +1,53 @@
-import React from "react";
+import axios from "axios";
+import { useState, useEffect } from "react";
 import ArticleBlock from "./ArticleBlock";
 
-function Articles({ articles, isLoading }) {
+function Articles() {
+  const [articles, setArticles] = useState([]);
+  const [articleId, setArticleId] = useState();
+  const [oneArticle, setOneArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    axios
+      .get("https://nc-news-api-qfui.onrender.com/api/articles")
+      .then(({ data }) => {
+        setArticles(data.articles);
+        setIsLoading(false);
+      });
+
+    articleId > 0 &&
+      axios
+        .get(`https://nc-news-api-qfui.onrender.com/api/articles/${articleId}`)
+        .then(({ data }) => {
+          setOneArticle(data.article);
+          setIsLoading(false);
+        });
+  }, [articleId]);
```

Comment:
Rather than doing this, it would be better to create a separate component for an individual article. As you have all of this logic in the same component at the moment, it means that either all of the articles or one specific article will be rendered. From a UI perspective this isn't great, as the route will be /articles regardless of what is happening. It would be nicer if you could provide someone with a separate URL that will take them to a specific article 👍